pkg/module/sync/runner_impl.go

```diff
@@ -159,38 +154,40 @@ func (r *RunnerImpl) updateModuleStatusFromExistingModules(modules common.Module
 	return updateRequired
 }
 
-func stateFromManifest(obj *manifestV1alpha1.Manifest) v1alpha1.State {
-	state := v1alpha1.State(obj.Status.State)
-	if state == "" {
-		return v1alpha1.StateProcessing
+func stateFromManifest(obj client.Object) v1alpha1.State {
+	switch manifest := obj.(type) {
+	case *manifestV1alpha1.Manifest:
+		return v1alpha1.State(manifest.Status.State)
+	default:
+		return v1alpha1.StateError
 	}
-	return state
 }
 
 func (r *RunnerImpl) deleteNoLongerExistingModuleStatus(ctx context.Context,
-	moduleStatusMap map[string]*v1alpha1.ModuleStatus, kyma *v1alpha1.Kyma,
+	moduleStatusMap map[int]*v1alpha1.ModuleStatus, kyma *v1alpha1.Kyma,
 ) bool {
 	updateRequired := false
 	moduleStatusArr := kyma.GetNoLongerExistingModuleStatus()
 	if len(moduleStatusArr) == 0 {
 		return false
 	}
-	for i := range moduleStatusArr {
-		moduleStatus := moduleStatusArr[i]
-		module := manifestV1alpha1.Manifest{}
+	for idx := range moduleStatusArr {
+		moduleStatus := moduleStatusArr[idx]
+		module := unstructured.Unstructured{}
+		module.SetGroupVersionKind(schema.GroupVersionKind(moduleStatus.GroupVersionKind))
 		module.SetName(moduleStatus.Name)
 		module.SetNamespace(moduleStatus.Namespace)
 		err := r.getModule(ctx, &module)
 		if errors.IsNotFound(err) {
 			updateRequired = true
-			delete(moduleStatusMap, moduleStatus.ModuleName)
+			delete(moduleStatusMap, idx)
 		}
 	}
 	kyma.Status.ModuleStatus = convertToNewmoduleStatus(moduleStatusMap)
 	return updateRequired
 }
 
-func convertToNewmoduleStatus(moduleStatusMap map[string]*v1alpha1.ModuleStatus) []v1alpha1.ModuleStatus {
+func convertToNewmoduleStatus(moduleStatusMap map[int]*v1alpha1.ModuleStatus) []v1alpha1.ModuleStatus {
 	newModuleStatus := make([]v1alpha1.ModuleStatus, 0)
```

Comment:
iterate through this map can't make sure original index keep same after it in the ModuleStatus array due to the go map is unordered.

Reply:
I believe the index in this case is unimportant since we just need 0(1) access. We could even just use an array but I didn't want to touch the types too much

Reply:
changed to array indexing